### ISSUE TITLE
Update InputCheckbox and InputRadio to ensure correct text alignment

### ISSUE THREAD

**Issue by clarsen-czi** [chanzuckerberg/sci-components#441]
### Issue:

A bug has been discovered in the InputCheckbox and InputRadio where when label text overflows onto additional lines the inputs are center-aligned to the text as opposed to remaining top-aligned to the first line of the label.

---

### Current behavior:

![Image](https://user-images.githubusercontent.com/87045841/235223392-d344789c-bb7d-4091-bbb6-d332b40702dd.png)

---

### Expected behavior:

![Image](https://user-images.githubusercontent.com/87045841/235223238-a77538de-18bb-4741-b996-e0bcfe98727c.png)

**Comment by cathystoli** [comment]
@liaprins did you follow up with Irene on this? Would this make sense for you to take over?

**Comment by liaprins** [comment]
> @liaprins did you follow up with Irene on this? Would this make sense for you to take over?

So sorry, I somehow missed seeing this message. I am trying to work on this now, though.

Here are some clues I'm documenting even though I haven't fully figured this out yet:
1.  The `<label>` element needs to receive styling of `align-items: start;` to keep checkbox at top. (e.g. add `align-items: start;` to styles.ts ~line 22, so it's added to the `StyledFormControlLabel` const)
2. However, this makes the text label too high relative to the checkbox/radio button at all times (whether or not text overflows to another line). I am trying to discern how to hold the text down lower, maybe by a top padding or margin. However, currently in styles.ts I could not find any variable or other reference to the text alone, only to the label as a whole (checkbox/radio button and text combined), and the caption.
3. I found a likely related issue that may make sense to fix at same time as this: The caption does not move down when main text goes to multiple lines, so it gets covered up by the main text. See screenshot below.

I started a branch called `441-inputcheckbox-inputradio-alignment` that has `align-items: start;` applied to the `<label>` element but no other changes yet.

![Screenshot 2024-01-12 at 3 58 03 PM](https://github.com/chanzuckerberg/sci-components/assets/17435353/a52585af-ae3f-4fca-8c63-7ca580b18875)
